src/main/thrift/parquet.thrift

```diff
@@ -552,6 +553,44 @@ struct DataPageHeaderV2 {
   8: optional Statistics statistics;
 }
 
+/** Block-based algorithm type annotation. **/
+struct SplitBlockAlgorithm {}
+
+/** The algorithm used in Bloom filter. **/
+union BloomFilterAlgorithm {
+  /** Block-based Bloom filter. **/
+   1: SplitBlockAlgorithm BLOCK;
+}
+
+/** Hash strategy type annotation. It uses Murmur3Hash_x64_128 from the original SMHasher
+ * repo by Austin Appleby.
```

Comment:
I don't think we should depend on a external codebase for this implementation (as it is subject to change). May be document the actual implementation details here instead of citing the reference.

Reply:
Thanks, we are citing a kind of well-known hash which is widely used in hive, orc, guava and etc..  It just like codec definition above we don't have implementation detail document. In the project hive, orc, and guava, I don't see they have implementation document.